src/TestInfrastructure/ToolWindowCommandTests.cs

```diff
@@ -31,13 +33,19 @@ namespace SonarLint.VisualStudio.Integration.UnitTests
     public abstract class ToolWindowCommandTests<T> where T : ToolWindowPane
     {
         private readonly Action<AsyncPackage, ILogger> executeCommand;
+        private readonly Func<IMenuCommandService, object> createCommand;
+        private readonly IDictionary<Guid, IEnumerable<int>> commandsInCommandSet;
 
         private Mock<AsyncPackage> package;
         private Mock<ILogger> logger;
 
-        protected ToolWindowCommandTests(Action<AsyncPackage, ILogger> executeCommand)
+        protected ToolWindowCommandTests(Action<AsyncPackage, ILogger> executeCommand, 
+            Func<IMenuCommandService, object> createCommand, 
+            IDictionary<Guid, IEnumerable<int>> commandsInCommandSet)
```

Comment:
Passing multiple delegates into the constructor is pretty ugly in general, and seems a particularly odd thing to do when it's an abstract class. Abstract methods/properties would look cleaner.